### PR TITLE
[KYUUBI #2375][FOLLOWUP] Implement BatchesResource GET /batches

### PR DESCRIPTION
**Pull request by zhaomin1423**
### _Why are the changes needed?_

To return all types of batch when empty batch type specified.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by zhaomin1423** [comment]
please help me to review it. @turboFei @ulysses-you

File: kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionManager.scala [hunk 1]
```diff
@@ -157,8 +157,9 @@ class KyuubiSessionManager private (name: String) extends SessionManager(name) {
 
   def getBatchSessionList(batchType: String, from: Int, size: Int): Seq[Session] = {
     allSessions().filter {
-      case batchSession: KyuubiBatchSessionImpl =>
+      case batchSession: KyuubiBatchSessionImpl if batchType != null =>
         batchSession.batchJobSubmissionOp.batchType.equalsIgnoreCase(batchType)
+      case _: KyuubiBatchSessionImpl => true
       case _ => false
     }.toSeq.sortBy(_.handle.identifier.toString).slice(from, from + size)
```
Comment (ulysses-you) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionManager.scala:164]: we can pull out the condition:

```
val sessions = if (isnull) all else ...
session.toSeq.sortBy..
```

Comment (turboFei) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionManager.scala:164]: > we can pull out the condition:
> 
> ```
> val sessions = if (isnull) all else ...
> session.toSeq.sortBy..
> ```

+1

File: kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionManager.scala [hunk 2]
```diff
@@ -157,8 +157,9 @@ class KyuubiSessionManager private (name: String) extends SessionManager(name) {
 
   def getBatchSessionList(batchType: String, from: Int, size: Int): Seq[Session] = {
     allSessions().filter {
-      case batchSession: KyuubiBatchSessionImpl =>
+      case batchSession: KyuubiBatchSessionImpl if batchType != null =>
```
Comment (turboFei) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionManager.scala:160]: by the way, I am not sure if no specified batchType, its default values is `null` or empty string?

```
Option(batchType).exists(_.nonEmpty)
```

File: kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionManager.scala (hunk 1 above)
Comment (zhaomin1423) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionManager.scala:164]: ignore from and size?

File: kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionManager.scala (hunk 2 above)
Comment (zhaomin1423) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionManager.scala:160]: it is null

File: kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionManager.scala (hunk 1 above)
Comment (zhaomin1423) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionManager.scala]: updated

**Review by ulysses-you — APPROVED** [review]

**Review by turboFei — APPROVED** [review]

**Comment by turboFei** [comment]
merging to master, thanks

**Pull request closed without merging**